src/model/entity/level.go

```diff
@@ -1,8 +1,76 @@
 package entity
 
+import (
+	"errors"
+	"math/rand"
+)
+
+const (
+	RoomsCount    = 9
+	RoomMinWidth  = 3
+	RoomMinHeight = 3
+	RoomWalls     = 2
+)
+
 type Level struct {
 	Rooms       []Room
 	Passages    []Passage
 	LevelNumber uint
 	LevelEnd    Box
 }
+
+func (l *Level) GenerateRoomsOnLevel(sizeMapWidth int, sizeMapHeight int) error {
+	l.Rooms = make([]Room, RoomsCount)
+
+	indices := rand.Perm(RoomsCount)
+	startIndex := indices[0]
+	finishIndex := indices[1]
+	var roomType RoomType
+
+	sizeSectionWidth := sizeMapWidth / 3
+	sizeSectionHeight := sizeMapHeight / 3
+
+	if sizeSectionWidth < RoomMinWidth || sizeSectionHeight < RoomMinHeight {
+		return errors.New("map size is too small: each room must be at least 3x3")
+	}
+
+	for i := 0; i < 3; i++ {
+		for j := 0; j < 3; j++ {
+			switch i*3 + j {
```

Comment:
Ещё можно переименовать абстрактные i, j в:
```suggestion
	for row := 0; row < 3; row++ {
		for col := 0; col < 3; col++ {
			switch row*3 + col {
```

Для понятности, что работа идёт именно с сеткой.